### ISSUE TITLE
Interaction between check, zero-warning, and fallback server address

### ISSUE THREAD

**Issue by ethanmdavidson** [haproxy/haproxy#2627]
### Detailed Description of the Problem

similar to #2176, except with a fallback address.

When running haproxy with `haproxy -c -dW -dr -f /usr/local/etc/haproxy/conf.d/`, check mode is failing due to server address resolution failures, despite `-dr` being set.

From the [haproxy manual](https://docs.haproxy.org/2.8/management.html#3):

> -c : only performs a check of the configuration files and exits before trying
    to bind. The exit status is zero if everything is OK, or non-zero if an
    error is encountered. Presence of warnings will be reported if any.

>  -dW : if set, haproxy will refuse to start if any warning was emitted while
   processing the configuration. This helps detect subtle mistakes and keep the
   configuration clean and portable across versions. It is recommended to set
   this option in service scripts when configurations are managed by humans,
   but it is recommended not to use it with generated configurations, which
   tend to emit more warnings. It may be combined with "-c" to cause warnings
   in checked configurations to fail. This is equivalent to global option
   "zero-warning".

> -dr : ignore server address resolution failures. It is very common when
    validating a configuration out of production not to have access to the same
    resolvers and to fail on server address resolution, making it difficult to
    test a configuration. This option simply appends the "none" method to the
    list of address resolution methods for all servers, ensuring that even if
    the libc fails to resolve an address, the startup sequence is not
    interrupted.

However, even with `-dr` set, the server address resolution failures are printed at `WARNING` level when a fallback server is used, causing check mode to fail, despite the fact that it is set to "ignore server address resolution failures"

example log message:
```
0.348 [WARNING]  (8) : config : [/usr/local/etc/haproxy/conf.d//local.cfg:136] : 'server example/example' : could not resolve address 'example.myinternaldomain', falling back to configured address.
```

### Expected Behavior

I would expect that when `-dr` is set, server address resolution failures would not cause check mode to fail, even if a fallback address is given.

### Steps to Reproduce the Behavior

1. Create a backend server with a fallback address specified using the `init-addr` directive, e.g. `server example example.myinternaldomain:80 init-addr last,libc,172.17.0.1`
2. run haproxy in check mode with `-c -dW -dr` flags

### Do you have an idea how to solve the issue?

Either don't print address fallback as WARNING when `-dr` is set, or continue printing them at WARNING but add a workaround in the check logic to ignore those specific warnings.

The resolution from #2176 was to print the resolution failures at NOTICE instead of WARNING.

### What is your configuration?

```haproxy
global
	log stdout format raw daemon notice
	log stderr format raw daemon warning
	user haproxy
	group haproxy

defaults http_defaults
	mode http
	log global
	option redispatch

defaults api_backend_defaults
	mode http
	log global
	default-server no-ssl
	option forwardfor

frontend api from http_defaults
	bind *:9001
	log stdout format raw daemon info
	option httplog

	use_backend example if { path_beg -i /example }

backend example from api_backend_defaults
	server example example.myinternaldomain:80 init-addr last,libc,172.17.0.1
```

### Output of `haproxy -vv`

```plain
HAProxy version 3.0.2-a45a8e6 2024/06/14 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2029.
Known bugs: http://www.haproxy.org/bugs/bugs-3.0.2.html
Running on: Linux 6.5.0-41-generic #41~22.04.2-Ubuntu SMP PREEMPT_DYNAMIC Mon Jun  3 11:32:55 UTC 2 x86_64
Build options :
  TARGET  = linux-glibc
  CC      = cc
  CFLAGS  = -O2 -g -fwrapv
  OPTIONS = USE_GETADDRINFO=1 USE_OPENSSL=1 USE_LUA=1 USE_PROMEX=1 USE_PCRE2=1 USE_PCRE2_JIT=1
  DEBUG   = 

Feature list : -51DEGREES +ACCEPT4 +BACKTRACE -CLOSEFROM +CPU_AFFINITY +CRYPT_H -DEVICEATLAS +DL -ENGINE +EPOLL -EVPORTS +GETADDRINFO -KQUEUE -LIBATOMIC +LIBCRYPT +LINUX_CAP +LINUX_SPLICE +LINUX_TPROXY +LUA +MATH -MEMORY_PROFILING +NETFILTER +NS -OBSOLETE_LINKER +OPENSSL -OPENSSL_AWSLC -OPENSSL_WOLFSSL -OT -PCRE +PCRE2 +PCRE2_JIT -PCRE_JIT +POLL +PRCTL -PROCCTL +PROMEX -PTHREAD_EMULATION -QUIC -QUIC_OPENSSL_COMPAT +RT +SHM_OPEN +SLZ +SSL -STATIC_PCRE -STATIC_PCRE2 +SYSTEMD +TFO +THREAD +THREAD_DUMP +TPROXY -WURFL -ZLIB

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_TGROUPS=16, MAX_THREADS=256, default=12).
Built with OpenSSL version : OpenSSL 3.0.11 19 Sep 2023
Running on OpenSSL version : OpenSSL 3.0.11 19 Sep 2023
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
OpenSSL providers loaded : default
Built with Lua version : Lua 5.4.4
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with libslz for stateless compression.
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.42 2022-12-11
PCRE2 library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with gcc compiler version 12.2.0

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
         h2 : mode=HTTP  side=FE|BE  mux=H2    flags=HTX|HOL_RISK|NO_UPG
  <default> : mode=HTTP  side=FE|BE  mux=H1    flags=HTX
         h1 : mode=HTTP  side=FE|BE  mux=H1    flags=HTX|NO_UPG
       fcgi : mode=HTTP  side=BE     mux=FCGI  flags=HTX|HOL_RISK|NO_UPG
  <default> : mode=TCP   side=FE|BE  mux=PASS  flags=
       none : mode=TCP   side=FE|BE  mux=PASS  flags=NO_UPG

Available services : prometheus-exporter
Available filters :
	[BWLIM] bwlim-in
	[BWLIM] bwlim-out
	[CACHE] cache
	[COMP] compression
	[FCGI] fcgi-app
	[SPOE] spoe
	[TRACE] trace
```

**Comment by capflam** [comment]
Indeed, it is not expected to have a warning in that case. The configuration is valid and it is the expected behavior. A Notice message must be emitted instead of a warning. And for me, it is unrelated to the `-dr` option. Because providing a fallback IP address is a valid use-case. Having the info that this fallback address is used during HAProxy startup may be useful, but it must not be a warning. I'll fix it. Thanks !

**Comment by capflam** [comment]
Fixed on 3.1-dev. Will be backported as usual.
